client/components/checklist/checklist.js

```diff
@@ -144,6 +137,9 @@ class Checklist extends PureComponent {
 	}
 }
 
-export default connect( ( state ) => ( {
-	siteId: getSelectedSiteId( state ),
-} ) )( localize( Checklist ) );
+const ChecklistForSite = ( props ) => {
+	const siteId = useSelector( getSelectedSiteId );
+	return <Checklist key={ siteId } siteId={ siteId } { ...props } />;
+};
+
+export default localize( ChecklistForSite );
```

Comment:
Nit: I know this kind of defeats the purpose of a component named `ChecklistForSite` but do you think it would make sense to move the injection of `translate` to that component as well? That would remove one layer of nesting in the components tree.

Reply:
Good call! Addressed in https://github.com/Automattic/wp-calypso/pull/60289/commits/60d79e06b4e8beeded1ca81dca7edc5867abb287